compiler/packages/babel-plugin-react-compiler/src/__tests__/fixtures/compiler/phi-type-inference-array-push-consecutive-phis.expect.md

```diff
@@ -0,0 +1,110 @@
+
+## Input
+
+```javascript
+import {makeArray} from 'shared-runtime';
+
+function Component(props) {
+  const x = {};
+  let y;
+  if (props.cond) {
+    if (props.cond2) {
+      y = [props.value];
+    } else {
+      y = [props.value2];
+    }
+  } else {
+    y = [];
+  }
+  // This should be inferred as `<store> y` s.t. `x` can still
+  // be independently memoized. *But* this also must properly
+  // extend the mutable range of the array literals in the
+  // if/else branches
+  y.push(x);
+
+  return [x, y];
+}
+
+export const FIXTURE_ENTRYPOINT = {
+  fn: Component,
+  params: [{cond: true, cond2: true, value: 42}],
+  sequentialRenders: [
+    {cond: true, cond2: true, value: 3.14},
+    {cond: true, cond2: true, value: 42},
+    {cond: true, cond2: true, value: 3.14},
+    {cond: true, cond2: false, value2: 3.14},
+    {cond: true, cond2: false, value2: 42},
+    {cond: true, cond2: false, value2: 3.14},
+    {cond: false},
+    {cond: false},
+  ],
+};
+
+```
+
+## Code
+
+```javascript
+import { c as _c } from "react/compiler-runtime";
+import { makeArray } from "shared-runtime";
+
+function Component(props) {
+  const $ = _c(3);
+  let t0;
+  if ($[0] === Symbol.for("react.memo_cache_sentinel")) {
+    t0 = {};
+    $[0] = t0;
+  } else {
+    t0 = $[0];
+  }
+  const x = t0;
+  let t1;
+  if ($[1] !== props) {
+    let y;
+    if (props.cond) {
+      if (props.cond2) {
+        y = [props.value];
+      } else {
+        y = [props.value2];
+      }
+    } else {
+      y = [];
+    }
+
+    y.push(x);
```

Comment:
while recognizing that all of these arrays being assigned to `y` are mutated at the `y.push(x)`